pkg/cache/observability/logger.go

```diff
@@ -0,0 +1,236 @@
+package observability
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+// ANSI Color Codes.
+const (
+	Reset  = "\033[0m"
+	Red    = "\033[31m"
+	Green  = "\033[32m"
+	Yellow = "\033[33m"
+	Blue   = "\033[34m"
+	Cyan   = "\033[36m"
+	Gray   = "\033[90m"
+)
+
+const (
+	INFO  = "INFO"
+	WARN  = "WARN"
+	ERROR = "ERROR"
+	DEBUG = "DEBUG"
+)
+
+const ansiRegex = "[\u001B\u009B][[]()#;?]*.{0,2}(?:(?:;\\d{1,3})*.[a-zA-Z\\d]|(?:\\d{1,4}/?)*[a-zA-Z])"
```

Comment:
You are using a regexp.Compile for this

Use it directly


```suggestion
var ansiRegex = regexp.MustCompile("[\u001B\u009B][[]()#;?]*.{0,2}(?:(?:;\\d{1,3})*.[a-zA-Z\\d]|(?:\\d{1,4}/?)*[a-zA-Z])")
```